types/stampit/index.d.ts

```diff
@@ -1,518 +1,560 @@
-         * Deeply merge a variable number of objects and add them to a new stamp and
-         * any future stamp it composes. Creates and returns a new Stamp. Chainable.
-         * @param deepStatics The object(s) containing static properties to be
-         * merged
-         * @returns A new stamp
-         */
-        deepStatics(...deepStatics: Array<{}>): Stamp;
-
-        /**
-         * Deeply merge a variable number of objects and add them to a new stamp and
-         * any future stamp it composes. Creates and returns a new Stamp. Chainable.
-         * @param deepStatics The object(s) containing static properties to be
-         * merged
-         * @returns A new stamp
-         */
-        staticDeepProperties(...deepStatics: Array<{}>): Stamp;
+    // TODO: Improve test by checking the type of common keys
+    type IsADescriptor<Type> = unknown extends Type
+        ? (keyof Type extends never
+              ? false
+              : keyof Type extends infer K
+              ? (K extends keyof Descriptor<unknown> ? true : false)
+              : false)
+        : false;
 
-        /**
-         * Shallowly assign properties of Stamp arbitrary metadata and add them to
-         * a new stamp and any future Stamp it composes. Creates and returns a new
-         * Stamp. Chainable.
-         * @param confs The object(s) containing metadata properties
-         * @returns A new Stamp
-         */
-        conf(...confs: Array<{}>): Stamp;
-
-        /**
-         * Shallowly assign properties of Stamp arbitrary metadata and add them to
-         * a new stamp and any future Stamp it composes. Creates and returns a new
-         * Stamp. Chainable.
-         * @param confs The object(s) containing metadata properties
-         * @returns A new Stamp
-         */
-        configuration(...confs: Array<{}>): Stamp;
-
-        /**
-         * Deeply merge properties of Stamp arbitrary metadata and add them to a new
-         * Stamp and any future Stamp it composes. Creates and returns a new Stamp.
-         * Chainable.
-         * @param deepConfs The object(s) containing metadata properties
-         * @returns A new Stamp
-         */
-        deepConf(...deepConfs: Array<{}>): Stamp;
-
-        /**
-         * Deeply merge properties of Stamp arbitrary metadata and add them to a new
-         * Stamp and any future Stamp it composes. Creates and returns a new Stamp.
-         * Chainable.
-         * @param deepConfs The object(s) containing metadata properties
-         * @returns A new Stamp
-         */
-        deepConfiguration(...deepConfs: Array<{}>): Stamp;
+    /**
+     * A function used as `.initializers` argument.
+     * @template Obj The type of the object instance being produced by the `Stamp`.
+     * @template S̤t̤a̤m̤p̤ The type of the `Stamp` producing the instance.
+     */
+    interface Initializer<Obj, S̤t̤a̤m̤p̤ extends StampSignature> {
+        // ??? should the type of `this` be the object or the stamp
```

Comment:
Answering the question. The `this` context of initializers is the object instance created from the stamp.